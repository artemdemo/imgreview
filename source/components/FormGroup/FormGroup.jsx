import React from 'react';

import './FormGroup.less';

const FormGroup = (props) => {
    return (
        <div className='form-group'>
            {props.children}
        </div>
    );
};

export default FormGroup;
