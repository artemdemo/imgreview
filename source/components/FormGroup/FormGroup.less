.form-group {
    margin-bottom: 1rem;
}

.form-group-error-text {
    position: absolute;
    color: #f44336;
}
