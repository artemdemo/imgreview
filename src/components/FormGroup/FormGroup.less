.FormGroup {
  margin-bottom: 1rem;
}

.FormGroup__Error {
  position: absolute;
  color: #f44336;
}
