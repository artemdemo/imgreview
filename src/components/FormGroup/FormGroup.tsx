import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

type TProps = {
  errorText?: string;
};

const FormGroupSty = styled.div`
  margin-bottom: 1rem;
`;

const FormGroupErrTextSty = styled.div`
  position: absolute;
  color: #f44336;
`;

class FormGroup extends React.PureComponent<TProps> {
  static propTypes = {
    errorText: PropTypes.string,
  };

  static defaultProps = {
    errorText: '',
  };

  renderError() {
    if (this.props.errorText !== '') {
      return <FormGroupErrTextSty>{this.props.errorText}</FormGroupErrTextSty>;
    }
    return null;
  }

  render() {
    return (
      <FormGroupSty>
        {this.props.children}
        {this.renderError()}
      </FormGroupSty>
    );
  }
}

export default FormGroup;
