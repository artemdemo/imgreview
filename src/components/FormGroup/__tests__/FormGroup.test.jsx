import React from 'react';
import renderer from 'react-test-renderer';
import FormGroup from '../FormGroup';

describe('FormGroup', () => {
    it('should render', () => {
        const tree = renderer.create(
            <FormGroup />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with children', () => {
        const tree = renderer.create(
            <FormGroup>
                Some child #1
                <div>
                    Some child #2
                </div>
            </FormGroup>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
