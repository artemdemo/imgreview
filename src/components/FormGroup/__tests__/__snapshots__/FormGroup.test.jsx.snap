// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup should render 1`] = `
<div
  className="sc-AxjAm dsYsxv"
/>
`;

exports[`FormGroup should render with children 1`] = `
<div
  className="sc-AxjAm dsYsxv"
>
  Some child #1
  <div>
    Some child #2
  </div>
</div>
`;
